tests/libc_newlib/main.c

```diff
@@ -58,15 +58,18 @@ static void test_newlib(void)
     /*
      * Be sure `iprintf` and `printf` are used when `newlib` is included as
      * they should be visible in the final elf file for compile time tests
+     *
+     * With llvm and samr21-xpro, I could not directly do 'printf == iprintf'.
+     * But doing `(printf - iprintf) == 0` correctly checked if they are equal.
      */
 
 #ifdef MODULE_NEWLIB
 #ifdef MODULE_NEWLIB_NANO
     /* Nano maps iprintf to printf */
-    TEST_ASSERT(iprintf == printf);
+    TEST_ASSERT_MESSAGE((printf - iprintf) == 0, "iprintf == printf");
```

Comment:
I think this output might be confusing. The message is given when `iprintf != printf`, so this is what should be printed.

Reply:
Hm, intuitively I agreed, but a plain ```TEST_ASSERT(<expression>)``` also prints just the failed expression, not the opposite. So all the ```TEST_ASSERT*``` have an implicit ```assertion "<msg>" failed```.

Reply:
True, but with e.g. TEST_ASSERT_EQUAL_INT it tells you "This is not as expected, this is what I expect" (faked with changing an expected value in `tests-core`:

```
.......................................
core_cib_tests.test_cib_get__overflow (tests/unittests/tests-core/tests-core-cib.c 48) exp 4 was 3
....................................................
run 91 failures 1
```

Reply:
> (do I make sense? :wink:)

Yes, at least I see (from the beginning btw) where you are coming from. I'm just arguing that for `TEST_ASSERT()` we can't really change what is happening. The CPP is just printing `#exp`, because it's the sane (and most efficient) thing to do. However, for `TEST_ASSERT_MESSAGE` we have more freedom, so we should use it and print a more intuitive output. But in the end it is purely based on taste of how to do it IMHO, so let's @cladmi decide, what he thinks is best for his test ;-).

Reply:
:+1:

Reply:
I prefer keeping the "default" embunit behavior, show the condition  that failed in the message on assert.

In master with the error I had:

```
018-09-25 14:45:58,501 - INFO # main(): This is RIOT! (Version: 2018.10-devel-855-g028bc-hal-HEAD)
2018-09-25 14:45:58,503 - INFO # Newlib/nano test
2018-09-25 14:45:58,503 - INFO # .
2018-09-25 14:45:58,509 - INFO # tests.test_newlib (tests/libc_newlib/main.c 66) iprintf == printf
2018-09-25 14:45:58,509 - INFO # 
2018-09-25 14:45:58,511 - INFO # run 1 failures 1
```

And I only changed the condition to `(iprintf- printf)==0` because it made it work for llvm but wanted to keep the same message.
With only TEST_ASSERT we would also have `(iprintf - printf) == 0` so for me would feel strange to do the opposite of the normal behavior.